src/app/cube/details/components/action-panel/action-panel.component.ts

```diff
@@ -142,26 +143,30 @@ export class ActionPanelComponent implements OnInit, OnDestroy {
         this.learningObject.id
       );
     }
-    try {
-      if (!this.userIsAuthor && this.isReleased) {
-        this.saved = this.libraryService.has(this.learningObject);
-
-        if (!this.saved) {
-          try {
-            await this.libraryService.addToLibrary(this.learningObject.cuid, this.learningObject.version);
-          } catch (e) {
-            if (e.status === 201) {
-              this.toaster.success('Successfully Added!', 'Learning Object added to your library');
-              this.saved = true;
-              this.animateSaves();
-            }
+    if (!this.userIsAuthor && this.isReleased) {
+      this.saved = this.libraryService.has(this.learningObject);
+
+      if (!this.saved) {
+        try {
+          await this.libraryService.addToLibrary(this.learningObject.cuid, this.learningObject.version);
+        } catch (err: any) {
+          if (err.status !== 201) {
+            this.toaster.error('Error!', 'There was an error adding to your library');
+            this.addingToLibrary = false;
+            this.changeDetectorRef.detectChanges();
+            return;
+          } else {
+            // Do nothing if status is 201.
           }
         }
       }
-    } catch (error) {
-      this.toaster.error('Error!', 'There was an error adding to your library');
     }
-    this.libraryService.getLibrary();
+
+    this.toaster.success('Successfully Added!', 'Learning Object added to your library');
+    this.saved = true;
+    this.animateSaves();
```

Comment:
If it doesn't break something else (and knowing this action panel component it might) I believe this can get moved up into the try where you add to the library so the success, saved state change, and animations are all together.